packages/block-editor/src/components/block-variation-picker/index.native.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import {
+	ScrollView,
+	View,
+	Text,
+	TouchableWithoutFeedback,
+	Platform,
+} from 'react-native';
+import { map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { withSelect, useDispatch } from '@wordpress/data';
+import { compose, usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { createBlock } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+import {
+	PanelBody,
+	BottomSheet,
+	UnsupportedFooterControl,
+} from '@wordpress/components';
+import { Icon, close } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import MenuItem from '../inserter/menu-item';
+import styles from './style.scss';
+
+function BlockVariationPicker( { isVisible, onClose, clientId, variations } ) {
+	const createBlocksFromInnerBlocksTemplate = ( innerBlocksTemplate ) => {
+		return map(
+			innerBlocksTemplate,
+			( [ name, attributes, innerBlocks = [] ] ) =>
+				createBlock(
+					name,
+					attributes,
+					createBlocksFromInnerBlocksTemplate( innerBlocks )
+				)
+		);
+	};
+	const { replaceInnerBlocks } = useDispatch( 'core/block-editor' );
+	const hitSlop = { top: 22, bottom: 22, left: 22, right: 22 };
+	const isIOS = Platform.OS === 'ios';
+
+	const cancelButtonStyle = usePreferredColorSchemeStyle(
+		styles.cancelButton,
+		styles.cancelButtonDark
+	);
+
+	const onVariationSelect = ( variation ) => {
+		replaceInnerBlocks(
+			clientId,
+			createBlocksFromInnerBlocksTemplate( variation.innerBlocks ),
+			false
+		);
+		onClose();
+	};
+
+	return (
+		<BottomSheet
+			isVisible={ isVisible }
+			onClose={ onClose }
+			title={ __( 'Select a layout' ) }
+			contentStyle={ styles.contentStyle }
+			leftButton={
```

Comment:
Hmmm, I'm wondering if it would be beneficial to use the `useMemo` here. Thanks to that we could avoid creating the leftButton when it's not needed.

```js
const leftButton = useMemo( () => <TouchableWithoutFeedbackonPress={ onClose }> ... </TouchableWithoutFeedbackonPress>, [ onClose, cancelButtonStyle ] );
```